src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -10,7 +10,7 @@
 public class Tag {
 
     public static final String MESSAGE_CONSTRAINTS = "Tags names should be alphanumeric";
-    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
 
     public final String tagName;
 
```

Comment:
Perhaps Tag class could be made into an abstract class since a Tag object will not be instantiated?